File: app/src/test/java/com/example/java_unittest_mitch/models/NoteTest.java
package com.example.java_unittest_mitch.models;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;


class NoteTest
{
    @Test
    public void test_equal_notes() {
        Note note1=new Note(1,"title","content","time");
        Note note2=new Note(1,"title","content","time");
        assertEquals(note1,note2);
    }

    @Test
    public void test_areNotequal_notes_differnetId() {
        Note note1=new Note(1,"title","content","time");
        Note note2=new Note(2,"title","content","time");
        //assertFalse(note1==note2);
        assertNotEquals(note1,note2);
    }

    @Test
    public void test_equal_notes_withDifferent_timestamps() {
        Note note1=new Note(1,"title","content","time1");
        Note note2=new Note(1,"title","content","time2");
        assertEquals(note1,note2);
    }
    @Test
    public void test_areNotequal_notes_differnetTitle() {
        Note note1=new Note(1,"title1","content","time");
        Note note2=new Note(1,"title2","content","time");
        assertNotEquals(note1,note2);
    }
    @Test
    public void test_areNotequal_notes_differnetContent() {
        Note note1=new Note(1,"title","content1","time");
        Note note2=new Note(1,"title","content2","time");
        assertNotEquals(note1,note2);
    }

}

File: app/src/test/java/com/example/java_unittest_mitch/models/NoteTest.kt
package com.codingwithmitch.unittesting2.models

import com.example.java_unittest_mitch.models.Note
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class NoteTest
{
    private val TIME_1="1,1,2023"
    private val TIME_2="1,2,2023"
    /*Compare two equal notes*/
    @Test
    internal fun is_Notes_Are_Equal_when_they_have_equal_data()
    {
        //Arrange
        val note1= Note("majid","you are going to be successful",TIME_1)
        note1.id=1
        val note2=Note("majid","you are going to be successful",TIME_2)
        note2.id=2
        //Act

        //Assert
        Assertions.assertNotEquals(note1,note2)
    }
 

    //-----------------------------------------------------
    /*Compare notes with 2 different ids*/
    @Test
    internal fun isEqualFunc_returnFalse_when_ids_are_not_the_same()
    {
        val note1=Note("majid","is working in netherland",TIME_1)
        note1.id=1
        val note2=Note("majid","is working in netherland",TIME_2)
        note2.id=2

        Assertions.assertNotEquals(note1,note2)

    }
    //-----------------------------------------------------



    /*Compare notes with 2 different content*/
    @Test
    internal fun isEqualFunc_returnTrue_when_content_areNotTheSame()
    {
        val note1=Note("majid","is working in netherland",TIME_1)
        note1.id=1
        val note2=Note("majid","is working in netherland very well",TIME_2)
        note2.id=1

        Assertions.assertEquals(note1,note2)
    }

    //-----------------------------------------------------

    /*Compare notes with 2 differnt timestamp*/
    @Test
    internal fun isEqualFuct_returnTrue_when_timestampsAreDifferent()
    {
        val note1=Note("majid","is working in netherland",TIME_1)
        note1.id=1
        val note2=Note("majid","is working in netherland",TIME_2)
        note2.id=1

        Assertions.assertEquals(note1,note2)

    }

    //-----------------------------------------------------
    /*Compare notes with 2 different Title*/
    @Test
    internal fun isEqualFunc_returnFalse_when_titlesAreDifferent()
    {
        val note1=Note("majid2851","is working in netherland",TIME_1)
        note1.id=1
        val note2=Note("majid","is working in netherland",TIME_2)
        note2.id=1
        Assertions.assertNotEquals(note1,note2)
    }

    //-----------------------------------------------------




}